### ISSUE TITLE
Logged in session does not have permission

### ISSUE THREAD

**Issue by chrisnonstop** [seansfkelley/synology-download-manager#134]
Hi @seansfkelley !
When trying to add new download task, I receive the following 2 error messages.
Thanks in advance for considering the issue which is blocking me from using the extension.
Cheers and happy new year :)

**1. Failed to add download: No default destination**
![dsmgr_failed](https://user-images.githubusercontent.com/59433321/71659516-377c2700-2d48-11ea-8998-a51fd635a850.png)

**2. The logged in session does not have permission**
![dsmgr_check](https://user-images.githubusercontent.com/59433321/71659501-23382a00-2d48-11ea-9ef4-195b2008968f.png)

However:

**1. the default download location is already set to my "downloads" shared folder in the Download Station settings**
![download_station_settings](https://user-images.githubusercontent.com/59433321/71659557-54b0f580-2d48-11ea-80fb-865b69387478.PNG)

**2. the user has read/write access to "downloads" shared folder**
![user_permissions](https://user-images.githubusercontent.com/59433321/71659574-5f6b8a80-2d48-11ea-9897-37fb6c0f582b.PNG)

**3. the connection is successful**
![dsmgr_settings](https://user-images.githubusercontent.com/59433321/71659587-6befe300-2d48-11ea-8760-22ee91cc92ec.PNG)

**4. For reference I already tried to resolve according to #12**
by rebooting the DS box, the computer, and restarting the browser.

### Versions
**Extension version**: 0.6.4
**Browser and version**: FF 71.0 64-bit 
**DSM version**: 6.2.2-24922 Update 4
**Download Station version**: 3.8.14-3543
**OS**: Win 10-x64 v1709

### Failure Logs
- Could not find a "Debugging Output" section in the FF extension settings page
- The DS Log Center reports that the user logged in successfully via DSM

**Comment by seansfkelley** [comment]
I'm guessing the "no default destination" is caused by the lack of permissions, so I'm going to focus on that instead.

In order, I would probably do:

1. Triple-check that you are using the same user in the extension that you've configured in the screenshots. I've made this mistake in the past.
2. Check through some other means that the file permissions are working correctly. I do this by connecting over `smb://` via my desktop to the file share and making sure I can see the appropriate files, but I guess you could also log into DSM using the user in question (rather than an admin) and navigating to the files through File Station.
3. Once you've verified it should work, refresh the session in the extension by hitting "Test Connection and Save". Even if you don't change anything, it'll logout and log back in.
4. You already did most of the nuclear option (rebooting, etc.), but you didn't try removing and reinstalling the extension. You could try that if (3) doesn't work.

Please let me know if those steps help! If they don't, then I'll have to think of something more involved to fix it...

**Comment by chrisnonstop** [comment]
It's solved now, thanks!
What helped was your suggestion to log into DSM using the dsm user.
What happened:
- For the "No default destination" issue, i was sure to had set it up but that was for my regular dsm user, not for the sdm extension user. I solved it by logging into DSM with the sdm user and setting it up in the download station settings.
- For the "logged in session does not have permission" issue, it was the sdm user which was missing permission to the FileStation app. It had permissions for triggering download station parameters but not for actually writing files. I solved it by allowing permission to "File Station"
